docs/ECO-METADATA.md

```diff
@@ -0,0 +1,46 @@
+# EVE meta-data server for app instances
+
+EVE provides an old method to serve cloud-init meta-data in the form of a read-only CDROM disk which is created on the fly when an app instance is booted if the API specifies userData/cipherData in the [AppInstanceConfig message](../api/proto/config/appconfig.proto).
+
+However, there is also a need to provide access to meta-data which might change while the app instance is running.
+
+The first data which is needed by applications is determining the external IP address of the edge node, so that when there is a portmap ACL in place the application instance can determine which IP plus port its peers should use to connect to it.
+
+The initial meta-data service provides merely that, but over time we expect to add the rest of the cloud-init content.
+
+## Schema
+
+There is no existing industry standard schema specifying a notion of an external IP; existing schemas contain public and private IP addresses but the external IP is a different thing necessitated by the internal NAT which EVE deploys for the local network instances.
+
+Thus this particular part of the meta-data uses a EVE-unique schema, which we do not expect for other meta-data information.
+
+The API endpoint is <http://169.254.169.254/eve/v1/network.json>
+
+The returned json contains
+
+- caller-ip: a string with the IP address and TCP port of the requesting app instance
+- external-ipv4: a string with the external IPv4 address
+- hostname: a string with the UUID hostname assigned to the app instance
```

Comment:
well see -- the hostname is actually standard

Reply:
Yes, the only useful thing is the external-ipv4. caller-ip is just for debugging.